cmd/kubeadm/app/phases/upgrade/postupgrade_test.go

```diff
@@ -169,6 +174,110 @@ func TestWriteKubeletConfigFiles(t *testing.T) {
 	}
 }
 
+func TestUnupgradedControlPlaneInstances(t *testing.T) {
+	testCases := []struct {
+		name          string
+		pods          []corev1.Pod
+		currentNode   string
+		expectedNodes []string
+		expectError   bool
+	}{
+		{
+			name: "one unupgraded instance",
+			pods: []corev1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "kube-apiserver-1",
+						Namespace: metav1.NamespaceSystem,
+						Labels: map[string]string{
+							"component": constants.KubeAPIServer,
+						},
+					},
+					Spec: corev1.PodSpec{
+						NodeName: "node-1",
+						Containers: []corev1.Container{
+							{Name: constants.KubeAPIServer, Image: "k8s.gcr.io/kube-apiserver:v1"},
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "kube-apiserver-2",
+						Namespace: metav1.NamespaceSystem,
+						Labels: map[string]string{
+							"component": constants.KubeAPIServer,
+						},
+					},
+					Spec: corev1.PodSpec{
+						NodeName: "node-2",
+						Containers: []corev1.Container{
+							{Name: constants.KubeAPIServer, Image: "k8s.gcr.io/kube-apiserver:v2"},
+						},
+					},
+				},
+			},
+			currentNode:   "node-1",
+			expectedNodes: []string{"node-2"},
+			expectError:   false,
+		},
+		{
+			name: "all upgraded instances",
+			pods: []corev1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "kube-apiserver-1",
+						Namespace: metav1.NamespaceSystem,
+						Labels: map[string]string{
+							"component": constants.KubeAPIServer,
+						},
+					},
+					Spec: corev1.PodSpec{
+						NodeName: "node-1",
+						Containers: []corev1.Container{
+							{Name: constants.KubeAPIServer, Image: "k8s.gcr.io/kube-apiserver:v1"},
+						},
+					},
+				},
+			},
+			currentNode:   "node-1",
+			expectedNodes: nil,
+			expectError:   false,
+		},
+		{
+			name:          "no kube-apiserver pods",
+			pods:          []corev1.Pod{},
+			currentNode:   "node-1",
+			expectedNodes: nil,
+			expectError:   true,
+		},
+	}
+
+	for _, tc := range testCases {
+		var runtimeObjs []runtime.Object
+		for i := range tc.pods {
+			runtimeObjs = append(runtimeObjs, &tc.pods[i])
+		}
+		client := fake.NewSimpleClientset(runtimeObjs...)
+
+		nodes, err := unupgradedControlPlaneInstances(client, tc.currentNode)
+
+		if tc.expectError {
+			if err == nil {
+				t.Fatalf("[%s] Expected an error, but got none", tc.name)
```

Comment:
once you use `t.Run(...` there is no need to print tc.name here and in other places.

Reply:
thanks. will fix soon.